drivers/flash/spi_nand.c

```diff
@@ -0,0 +1,1034 @@
+/*
+ * Copyright (c) 2022 Macronix International Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT jedec_spi_nand
+
+#include <errno.h>
+#include <drivers/flash.h>
```

Comment:
The includes are not up to date with the new system

You need to change it like this
```suggestion
#include <zephyr/drivers/flash.h>
```